pkg/runtime/session.go

```diff
@@ -13,16 +13,19 @@
 
 package runtime
 
-import "github.com/aws/aws-sdk-go/aws/session"
+import (
+	ackv1alpha1 "github.com/aws/aws-controllers-k8s/apis/core/v1alpha1"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
 
-func NewSession() (*session.Session, error) {
-	// NOTE(jaypipes): session.NewSession() is needed for the STS::AssumeRole
-	// stuff we will need to do...
-	sess, err := session.NewSession()
-	if err != nil {
-		return nil, err
-	}
-	// TODO(jaypipes): Handling all common region endpoint, throttling
-	// configuration, TLS, etc
+func NewSession(region ackv1alpha1.AWSRegion) (*session.Session, error) {
+	// NOTE(jaypipes): session.must() is needed for the STS::AssumeRole
```

Comment:
why is `session.Must()` required for STS::AssumeRole, versus `session.NewSession()`?

The problem with `Must()` is that it will panic upon errors instead of returning the error "upstream" to the caller...